index.js

```diff
@@ -122,31 +124,73 @@ module.exports = class Config {
       }
       filePath = path.resolve(relativePath)
     }
+    else if (options.path) {
+      filePath = path.resolve(options.path)
+    }
+    else {
+      throw new Error('No file path was specified')
+    }
 
     const fileExtension = path
       .extname(filePath)
       .slice(1)
-    const fileContent = fs.readFileSync(filePath)
-    let configObject
+    let fileContent
 
-    if (fileExtension === 'json' || fileExtension === '') {
-      configObject = JSON.parse(fileContent)
+    try {
+      fileContent = fs.readFileSync(filePath)
     }
-    if (fileExtension === 'js') {
-      configObject = require(filePath)
+    catch (error) {
+      const notExistant = error.message.includes('no such file')
+      const isDirectory = error.message
+        .includes('illegal operation on a directory')
+
+      if (isRequired) throw error
+
+      if (notExistant) {
+        return
+      }
+      else if (isDirectory) {
+        if (shallPrintWarning) {
+          console.warn(`Warning: Tried to load the directory "${filePath}"`)
+        }
+        return
+      }
+      else {
+        throw error
+      }
     }
-    else if (fileExtension === 'yaml') {
-      configObject = yaml.safeLoad(fileContent)
+
+    let configObject
+
+    try {
+      if (fileExtension === 'json' || fileExtension === '') {
+        configObject = JSON.parse(fileContent)
+      }
+      else if (fileExtension === 'js') {
+        configObject = require(filePath)
+      }
+      else if (fileExtension === 'yaml') {
+        configObject = yaml.safeLoad(fileContent)
+      }
+      else {
+        throw new TypeError(`"${fileExtension}" is no supported file extension`)
+      }
     }
-    else {
-      throw new TypeError(`"${fileExtension}" is no supported file extension`)
+    catch (error) {
+      console.error(
```

Comment:
Why do you log the error, when you rethrow it anyway? Logging should be decided by the function that catches the error.

Reply:
The error message wasn't helpful, but you're right, actually the error message should be extended.